### ISSUE TITLE
Rule Change: Make `sort-imports` `memberSyntaxSortOrder` more flexible

### ISSUE THREAD

**Issue by jack5github** [eslint/eslint#20791]
### What rule do you want to change?

sort-imports

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
import { IWorld } from '@cucumber/cucumber'; // Installed package starting with 'c'
import assert from 'node:assert'; // Installed package starting with 'n'
import * as meta from './meta.js'; // Script file in '.'
import { assertObjType, convertObj, locateObj } from '../utils.js'; // Script file in '..'
```

### What does the rule currently do for this code?

`sort-imports` is incompatible with the sorted imports in the style of the example code. They are sorted by installed packages first, then script files, both alphabetically in their respective groups. This is true for many other code files, resulting in imports arranged in various ways, meaning that the member syntax groups are never truly lined up. The strict requirements of the `memberSyntaxSortOrder` property means that  `sort-imports` cannot be applied to the codebase, as it enforces ordering by member syntax groups, creating errors in a variety of files.

### What will the rule do after it's changed?

Allow `memberSyntaxSortOrder` to omit elements and (optionally) specify a wildcard (*). This enables lazy sorting of member syntax groups, or the complete disabling of member syntax sort altogether. The benefit of this change is that `sort-imports` will still be usable for alphabetically sorting members, but will not need to enforce line order beyond that.

```ts
// Required at minimum:
["none", "all", "multiple", "single"] // No change
["none", "all", "single"] // "multiple" will be expected after others
["none", "all"] // "single" or "multiple" will be expected after others
["all"] // All except "all" will be expected after
[] // Member syntax groups are not sorted
// Optional:
["none", "all", "multiple", "single", "*"] // No change
["*", "all", "multiple", "single"] // "none" will be expected before others
["*", "single", "multiple"] // "none" or "all" will be expected before others
["*", "multiple"] // All except "multiple" will be expected before
["*"] // Member syntax groups are not sorted
["none", "all", "*", "multiple"] // "single" will be expected between others
["none", "*", "multiple"] // "all" or "single" will be expected between others
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
The rule `sort-imports` is [frozen](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules), as such it does not accept feature requests anymore.
For your case you can use the option `ignoreDeclarationSort: true` and use a third-party plugin, e.g. [eslint-plugin-import](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md) or [eslint-plugin-perfectionist](https://perfectionist.dev/rules/sort-imports).
